fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/autobrakes.rs

```diff
@@ -541,5 +604,777 @@ impl SimulationElement for A380AutobrakeController {
+
+    const MAX_DECEL_SAFETY_MARGIN_RATIO: f64 = 1.4;
+    const MIN_DECEL_SAFETY_MARGIN_RATIO: f64 = 1.15;
+    const DECEL_SAFETY_MARGIN_SHAPING_FACTOR: f64 = 0.4;
+
+    const REMAINING_BRAKING_DISTANCE_END_OF_RUNWAY_OFFSET_METERS: f64 = 100.;
+
+    fn new(context: &mut InitContext) -> Self {
+        Self {
+            in_flight_btv_stopping_distance_id: context
+                .get_identifier("OANS_BTV_REQ_STOPPING_DISTANCE".to_owned()),
+            runway_length_id: context.get_identifier("OANS_RWY_LENGTH".to_owned()),
+            distance_to_exit_id: context
+                .get_identifier("OANS_BTV_REMAINING_DIST_TO_EXIT".to_owned()),
+            rot_estimation_id: context.get_identifier("BTV_ROT".to_owned()),
+
+            runway_length: Arinc429Word::new(Length::default(), SignStatus::NoComputedData),
+            rolling_distance: Length::default(),
+            in_flight_btv_stopping_distance: Arinc429Word::new(
+                Length::default(),
+                SignStatus::NoComputedData,
+            ),
+            oans_distance_to_exit: Arinc429Word::new(Length::default(), SignStatus::NoComputedData),
+
+            spoilers_active: false,
+
+            state: BTVState::Disabled,
+
+            deceleration_request: Acceleration::default(),
+            end_of_decel_acceleration: Acceleration::default(),
+            desired_deceleration: Acceleration::new::<meter_per_second_squared>(
+                Self::MAX_DECEL_DRY_MS2,
+            ),
+            actual_deceleration: Acceleration::default(),
+
+            final_distance_remaining: Length::default(),
+
+            distance_remaining_at_decel_activation: Length::default(),
+
+            dry_prediction: Length::default(),
+            wet_prediction: Length::default(),
+
+            distance_to_rwy_end: Length::default(),
+        }
+    }
+
+    fn enable(&mut self) {
+        if self.state == BTVState::Disabled && self.arming_authorized() {
+            self.state = BTVState::Armed;
+        }
+    }
+
+    fn disarm(&mut self) {
+        self.state = BTVState::Disabled;
+        self.deceleration_request = Acceleration::new::<meter_per_second_squared>(5.);
+        self.end_of_decel_acceleration = Acceleration::new::<meter_per_second_squared>(5.);
+        self.final_distance_remaining = Length::default();
+        self.distance_remaining_at_decel_activation = Length::default();
+        self.desired_deceleration =
+            Acceleration::new::<meter_per_second_squared>(Self::MAX_DECEL_DRY_MS2);
+    }
+
+    fn decel(&self) -> Acceleration {
+        match self.state {
+            BTVState::Decel | BTVState::_OutOfDecelRange => self.deceleration_request,
+            BTVState::EndOfBraking => self.end_of_decel_acceleration,
+            BTVState::RotOptimization => self.accel_during_rot_opti(),
+            _ => Acceleration::new::<meter_per_second_squared>(5.),
```

Comment:
This is against coding guideline: https://github.com/flybywiresim/aircraft/blob/master/fbw-common/src/wasm/systems/guidelines.md#explicit-match-pattern-declaration